packages/application/src/api/json-rpc/responseHandlers.ts

```diff
@@ -128,15 +129,46 @@ const hasRequiredProps = <T extends Record<string, unknown>>(
 	return ok(obj)
 }
 
-export const handleTransactionHistoryResponse = executedTXDecoders.create<
-	TransactionHistoryEndpoint.Response,
-	TransactionHistoryEndpoint.DecodedResponse
->()
+export const handleTransactionHistoryResponse = (
+	json: TransactionHistoryEndpoint.Response,
+) =>
+	executedTXDecoders
+		.create<
+			TransactionHistoryEndpoint.Response,
+			TransactionHistoryEndpoint.DecodedResponse
+		>()(json)
+		.andThen(decoded =>
+			ok({
+				...decoded,
+				transactions: decoded.transactions.map(tx => ({
+					...tx,
+					message: tx.message
+						? Message.plaintextToString(
```

Comment:
Hmm this is wrong right? Here you assume that ALL messages are plaintext? I think you are missing one check first, to check if it is encrypted, then don't change, but if plaintext, then perform this?

Reply:
oops